src/main/scala/org/broadinstitute/dsde/workbench/sam/api/SecurityDirectives.scala

```diff
@@ -51,6 +51,33 @@ trait SecurityDirectives {
       }
     }
 
+  def requireCreateWithParent(maybeParent: Option[FullyQualifiedResourceId], resourceType: ResourceType, userId: WorkbenchUserId, samRequestContext: SamRequestContext): Directive0 = {
+    maybeParent match {
+      case None => Directives.pass // no parent specified, proceed
+      case Some(parent) =>
+        // parents are allowed for a resource type if the owner role contains the SamResourceActions.setParent action
+        val parentAllowed = resourceType.roles.find(_.roleName == resourceType.ownerRoleName).exists(_.actions.contains(SamResourceActions.setParent))
```

Comment:
Is this just checking if the resource has parent options available? It seems weird to have a permission on a child to be whether you can set the parent or not.

Reply:
why is that weird?

Reply:
I would expect that to be a permission on the parent, not on the child. (And then it checks if there are parents available, and if there are none, its irrelevant.)

Unless this is about like… switching billing projects for a workspace or something? (Or moving a notebook to another workspace or something? I could see that being a thing, but it seems like it might be better represented as permissions on the parent)

Reply:
Resolved in voice.